contracts/Governance.sol

```diff
@@ -0,0 +1,480 @@
+  uint256 public constant INIT_MIN_EXECUTE_SUPPORT_RATE = 50;
+  uint256 public constant PROPOSAL_EXECUTE_SUPPORT_RATE_SCALE = 100;
+
+  // locker => share contract => LockShare
+  mapping(address => mapping(address => ShareLock)) private lockShareMap;
+
+  Vote[] public votes;
+  // for proposal
+  ParamChangeRequest[] public paramChangeRequests;
+  Proposal[] public proposals;
+
+  uint256 public votingPeriod;
+  uint256 public executionDelay;
+  uint256 public executionExpiration;
+  uint256 public quorumVotingPower;
+  uint256 public minExecutableSupportRate;
+
+  // for poll
+  Poll[] public polls;
+  uint256 public pollVotingPeriod;
+  uint256 public pollSubmitThreshold;
+
+  enum ProposalState { Pending, Active, Defeated, Canceled, Timelocked, AwaitingExecution, Executed, Expired }
+  struct ParamChangeRequest {
+    string key;
+    bytes value;
+    address target;
+  }
+
+  struct Vote {
+    address voter;
+    uint256 votingPower;
+    bool support;
+  }
+
+  struct Proposal {
+    uint256[] requestIds;
+    uint256[] voteIds;
+
+    address proposer;
+    string description;
+
+    // vote starts at
+    uint256 startAt;
+    // vote ends at
+    uint256 endAt;
+    uint256 forVotingPower;
+    uint256 againstVotingPower;
+    bool executed;
+    bool canceled;
+  }
+
+  struct Poll {
+    uint256[] voteIds;
+
+    address proposer;
+    string description;
+
+    // vote starts at
+    uint256 startAt;
+    // vote ends at
+    uint256 endAt;
+    uint256 forVotingPower;
+    uint256 againstVotingPower;
+  }
+
+  struct ShareLock {
+    uint256 amount;
+    uint256 votingPower;
+
+    uint256[] votedProposalIds;
+    uint256[] votedPollIds;
+  }
+
+  event paramChange(string key, bytes value);
+  event ProposalCreated(
+    uint256 indexed id,
+
+    address indexed proposer,
+    string description,
+    uint256 startAt,
+    uint256 endAt
+  );
+  event ProposalExecuted(uint256 indexed proposalId, address indexed executor);
+  event ProposalVoted(uint256 indexed proposalId, address indexed voter, bool indexed support, address shareContract, uint256 shareAmount, uint256 votingPower);
+
+  event PollCreated(
+    uint256 indexed id,
+
+    address indexed proposer,
+    string description,
+    uint256 startAt,
+    uint256 endAt
+  );
+  event PollVoted(uint256 indexed proposalId, address indexed voter, bool indexed support, address shareContract, uint256 shareAmount, uint256 votingPower);
+
+  modifier onlyCabinet() {
+    uint256 indexPlus = IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).currentValidatorSetMap(msg.sender);
+    uint256 numOfCabinets = IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).numOfCabinets();
+    if (numOfCabinets == 0) {
+      numOfCabinets = 21;
+    }
+
+    require(indexPlus > 0 && indexPlus <= numOfCabinets, "not cabinet");
+    _;
+  }
+
+  function submitProposal(ParamChangeRequest[] memory requests, string memory _description, uint256 voteAt) public onlyCabinet {
```

Comment:
I hope anyone can submit a proposal, but with a bar, like the proposer must vote his proposal with some stBNB.

Reply:
add `executableProposalThreshold` for submitting executable proposal, remove `onlyCabinet`